server/datastore/mysql/migrations/data/20171212182459_DeleteSoftDeletedPacks.go

```diff
@@ -1,23 +0,0 @@
-package data
```

Comment:
Deleting migrations that have been published in a release can lead to inconsistencies. We need to leave these migrations and assume they have been run prior to the new migration running.

Reply:
Ah, this really confuses me, based on how I read the the `database-migrations.md` doc.

The section about 'adding/updating tables' says that:

> Once committed to the Fleet repo, table migrations should be considered immutable. Any changes to an existing table should take place in a new migration executing ALTERs.

However, the section on **data** migrations says:

> Data migrations can be mutable. If tables are altered in a way that would render a data migration invalid (columns changed/removed), data migrations should be updated to comply with the new schema.

So, to me, if the `deleted` / `deleted_at` columns no longer exist, that means the migration in the two packs are no longer valid and need to be updated. But it's less clear what that means as far as "updating to comply with the new schema" means.

I'm still not super clear on what I should be doing here instead however.

Reply:
Ah, no, you are correct. I had missed that these are data migrations that are no longer valid after the schema changes. Thank you for pushing back.